### PR TITLE
API Reference for Models, Messages, Threads

### PR DESCRIPTION
**Pull request by hahuyhoang411**
Fixes: #686

- [x] Model API
- [x] Messages API
- [x] Threads API

### CODE REVIEW

**Comment by hahuyhoang411** [comment]
This is only the draft of the Model API.

File: docs/openapi/specs/messages.yaml [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+                    items:
+                      type: string
+                    description: "Annotations for the text content, if any."
+        metadata:
+          type: object
+          description: "Metadata associated with the message, defaults to an empty object."
+        chat_completion_id:
+          type: string
+          description: "Identifier of the chat completion, if applicable."
+      required:
```
Comment (0xSage) [inline docs/openapi/specs/messages.yaml:55]: none of these are required fields.

File: docs/openapi/specs/messages.yaml [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        assistant_id:
+          type: string
+          description: "Identifier of the assistant involved in the message, if applicable."
+        run_id:
+          type: string
+          description: "Run ID associated with the message, if applicable."
+        metadata:
+          type: object
+          description: "Metadata associated with the message."
+      required:
```
Comment (0xSage) [inline docs/openapi/specs/messages.yaml:121]: not required

File: docs/openapi/specs/models.yaml [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+          additionalProperties: false
+        metadata:
+          type: object
+          description: "Additional metadata."
+        assets:
+          type: array
+          items:
+            type: string
+          description: "List of assets related to the model."
+      required:
```
Comment (0xSage) [inline docs/openapi/specs/models.yaml:112]: only required field is source_url

File: docs/openapi/specs/threads.yaml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+          description: "List of assistants involved in the thread."
+        metadata:
+          type: object
+          description: "Metadata associated with the thread."
+        messages:
+          type: array
+          items:
+            type: string
+          description: "List of messages within the thread."
+      required:
```
Comment (0xSage) [inline docs/openapi/specs/threads.yaml:77]: none of these fields are required

**Review by 0xSage — CHANGES_REQUESTED** [review]
1. Omit the docs/specs changes please. I'm working on that in a separate unmerged branch #681. Adding those unmerged changes in this PR will break my branch - this is not best practice. This PR should only update `/openapi/specs`. 

2. Typo in the title of this PR `Api refeference`

3. Majority fields are not required. Please read the docs/specs carefully - we specify which fields are required/not

**Comment by dan-jan** [comment]
@hahuyhoang411 @hieu-jan: Updating the title of the PR (please observe naming hygiene!)

**Review by 0xSage — APPROVED** [review]
woot

**Pull request merged**